### ISSUE TITLE
Can't ensure latest if package is not instatlled

### ISSUE THREAD

**Issue by nibalizer** [voxpupuli/puppet-nodejs#158]
Using the nodejs::npm type I have the following experience. When setting ensure => latest on a package that is not installed, I get the following error:

```puppet
package { 'express':
  ensure   => 'latest',
  provider => 'npm',
}

```
```
Error: Could not update: Got nil value for ensure
Error: /Stage[main]/Main/Package[express]/ensure: change from absent to latest failed: Could not update: Got nil value for ensure
```
Changing to:

```puppet
package { 'express':
  ensure   => 'present',
  provider => 'npm',
}
```
I can then successfully install it. Then changing it to latest and running again works.

**Comment by nibalizer** [comment]
I'm pretty sure the issue is in 'npm outdated -g <thingy>' because this doesn't print anything if the package is not installed. 

npm view <thingy> version prints the latest stable version whether the package is installed or not.

**Comment by waynr** [comment]
I just ran into this also, I opted to pin to a concrete version which seems to work fine.

**Comment by joelgarboden** [comment]
According to my research and testing (using Puppet 3.8.2).

Changing the ```latest``` definition to
```
  def latest
    npm('view', resource[:name], "version").strip
  end
```

And changing ```update``` to 

```
 def update
    self.install
 end
```

Resolves this issue, I tested extensively with the ```tar``` npm package. Present, latest, absent, explicit version, downgrades, and upgrades all work.

**Comment by igalic** [comment]
@joelgarboden would you like to make a patch for that?

**Comment by joelgarboden** [comment]
@igalic https://github.com/voxpupuli/puppet-nodejs/pull/199

Looks like tests pass, except for a robocop setting warning for Puppet 4.0, which I believe are unrelated to my changes.

**Comment by juniorsysadmin** [comment]
Fixed in #199
